packages/core/src/types.ts

```diff
@@ -1275,18 +1275,25 @@ export interface ActorRef<TEvent extends EventObject, TEmitted = any>
   send: Sender<TEvent>;
 }
 
-export interface SpawnedActorRef<TEvent extends EventObject, TEmitted = any>
-  extends ActorRef<TEvent, TEmitted> {
+export interface SpawnedActorRef<
+  TEvent extends EventObject,
+  TEmitted = any,
```

Comment:
What is `TEmitted` conceptually? I thought it was kinda a snapshot - are those output events? If yes - shouldn't this extend `EventObject`?

Reply:
`TEmitted` represents whatever is emitted to observers in `actor.subscribe(...)`. It doesn't necessarily extend `EventObject` since e.g., a Promise can emit non-event values.

Reply:
Right, but for promises that value that is emitted is the snapshot so I wonder in what situations the emitted value and snapshot would be different.

Reply:
The snapshot is the emitted value _at any point in time_. For observables and promises, that may be `undefined` if nothing was emitted yet. For machines, that is always defined (`State`) since the initial state is known immediately. That's why `TEmitted` and `TSnapshot` can be different.

Reply:
Removed the 3rd generic param in 1321351

Reply:
> That's why TEmitted and TSnapshot can be different.

That would mean that `getSnapshot` would not simply return `TSnapshot` for promises etc - since if your snapshot is a `number` then `getSnapshot` would have to return `number | undefined` (resolved from `TSnapshot | undefined`). That was my main concern about this.

Since you have unified them now - we could unify the naming further in the future and rename most `TEmitted` to `TSnapshot`, right? This would be mostly an internal change.

Reply:
> > That's why TEmitted and TSnapshot can be different.
> 
> That would mean that `getSnapshot` would not simply return `TSnapshot` for promises etc - since if your snapshot is a `number` then `getSnapshot` would have to return `number | undefined` (resolved from `TSnapshot | undefined`). That was my main concern about this.
> 
> Since you have unified them now - we could unify the naming further in the future and rename most `TEmitted` to `TSnapshot`, right? This would be mostly an internal change.

Added here: https://github.com/davidkpiano/xstate/projects/1#card-61626771